File: src/leetcode/2026/06/Solution1071.cpp
#include <string>
using namespace std;

class Solution1071 {
public:
    string gcdOfStrings(string str1, string str2) {
        if (str1 + str2 != str2 + str1) {
            return "";
        }
        int gcdLength = gcd(str1.size(), str2.size());
        return str1.substr(0, gcdLength);
    }

    int gcd(int a, int b) {
        return b == 0 ? a : gcd(b, a % b);
    }
};

int main() {
    Solution1071 solution;

    string str1 = "ABCABC";
    string str2 = "ABC";
    string result = solution.gcdOfStrings(str1, str2);
    printf("GCD of strings: %s\n", result.c_str());

    str1 = "ABABAB";
    str2 = "ABAB";
    result = solution.gcdOfStrings(str1, str2);
    printf("GCD of strings: %s\n", result.c_str());

    str1 = "LEET";
    str2 = "CODE";
    result = solution.gcdOfStrings(str1, str2);
    printf("GCD of strings: %s\n", result.c_str());

    str1 = "AAAAAB";
    str2 = "AAA";
    result = solution.gcdOfStrings(str1, str2);
    printf("GCD of strings: %s\n", result.c_str());

    return 0;
}

File: src/leetcode/2026/06/Solution1071.java
public class Solution1071 {
    public String gcdOfStrings(String str1, String str2) {
        if (!(str1 + str2).equals(str2 + str1)) {
            return "";
        }
        int gcdLength = gcd(str1.length(), str2.length());
        return str1.substring(0, gcdLength);
    }

    private int gcd(int a, int b) {
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }

    public static void main(String[] args) {
        Solution1071 solution = new Solution1071();

        String str1_1 = "ABCABC";
        String str2_1 = "ABC";
        String result1 = solution.gcdOfStrings(str1_1, str2_1);
        System.out.println(result1); // Output: "ABC"

        String str1_2 = "ABABAB";
        String str2_2 = "ABAB";
        String result2 = solution.gcdOfStrings(str1_2, str2_2);
        System.out.println(result2); // Output: "AB"

        String str1_3 = "LEET";
        String str2_3 = "CODE";
        String result3 = solution.gcdOfStrings(str1_3, str2_3);
        System.out.println(result3); // Output: ""

        String str1_4 = "AAAAAB";
        String str2_4 = "AAA";
        String result4 = solution.gcdOfStrings(str1_4, str2_4);
        System.out.println(result4); // Output: ""
    }
}
